extends Area3D

@export var brzina = 50 # PoveÄ‡aj brzinu za raketu
@export var explosion_scene: PackedScene 

@onready var ray = $RayCast3D

func _physics_process(delta):
	# 1. Pomakni raketu
	global_position -= transform.basis.z * brzina * delta
	
	# 2. Provjeri sudare
	if ray.is_colliding():
		var pogodjeni_objekt = ray.get_collider()
		
		# --- POPRAVAK OVDJE ---
		# Provjeravamo sadrÅ¾i li ime "Player" ili "vojnik"
		if "Player" in pogodjeni_objekt.name or "vojnik" in pogodjeni_objekt.name or pogodjeni_objekt.is_in_group("player"):
			
			# OVO JE KLJUÄŒNO: KaÅ¾emo RayCastu da ovaj objekt (igraÄa)
			# doda na listu iznimki. RayCast Ä‡e od iduÄ‡eg frame-a
			# prolaziti KROZ igraÄa kao da ne postoji.
			ray.add_exception(pogodjeni_objekt)
			
			return # Prekini funkciju, nemoj eksplodirati
			
		# Ako nije igraÄ, onda eksplodiraj
		print("ğŸš§ RayCast detektirao cilj: ", pogodjeni_objekt.name)
		detonate(ray.get_collision_point())

func detonate(point):
	print("ğŸ”¥ BOOM!")
	if explosion_scene:
		var expl = explosion_scene.instantiate()
		get_tree().root.add_child(expl)
		
		# Postavi eksploziju na toÄku sudara
		expl.global_position = point
	
	queue_free()
